crates/sui-core/src/authority.rs

```diff
@@ -1846,7 +1843,14 @@ impl AuthorityState {
                         Some(object) => {
                             let layout = object.get_layout(
                                 ObjectFormatOptions::default(),
-                                self.module_cache.as_ref(),
+                                // threading the epoch_store through this API does not
+                                // seem possible, so we just read it from the state (self) and fetch
+                                // the module cache out of it.
+                                // Notice that no matter what module cache we get things
+                                // should work
+                                self.load_epoch_store_one_call_per_task()
```

Comment:
nit: seems like elsewhere you just directly call `self.epoch_store.load()` which is certainly less alarming than this function call, but I'll leave it to @mystenmark or @lxfind to advise on what the best way is to make the store available here.

Reply:
yeah this is a pretty sore subject :) 
I liked the scaring side of that...

Reply:
so to conclude on this, I liked the more verbose API, so unless there are significant reservation I would leave as is